packages/components/src/disabled/index.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import type { RefCallback } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useDisabled } from '@wordpress/compose';
+import { createContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { StyledWrapper } from './styles/disabled-styles';
+import type { DisabledProps } from './types';
+import type { WordPressComponentProps } from '../ui/context';
+
+const Context = createContext< boolean >( false );
+const { Consumer, Provider } = Context;
+
+/**
+ * `Disabled` is a component which disables descendant tabbable elements and prevents pointer interaction.
+ *
+ * @example
+ * ```js
```

Comment:
Lets specify the example as `jsx` code

```suggestion
 * ```jsx
```